resources/migrations/000_migrations.yaml

```diff
@@ -10225,6 +10225,44 @@ databaseChangeLog:
                   'type/Title'
                 );
 
+  - changeSet:
+      id: v43.00-027
+      author: camsaul
+      comment: Added 0.43.0. Drop NOT NULL constraint for core_user.password
```

Comment:
We actually don't really even need that `password_salt` column since the `bcrypt` tooling we're using actually does it's own salting -- see https://github.com/metabase/metabase/issues/778. So worst case scenario if we don't set it things will still be salted.

Either way I think a `CHECK` constraint would totally make sense, but MySQL 5.7 doesn't support `CHECK` constraints. We **could** add one for Postgres, H2, and newer MySQL/MariaDB but I'm not sure it's worth all the hassle and conditional preconditions in the migration to figure out how to make it only happen for some DBs

Reply:
I would say we should still add it anyway for the DBs that do support it if it weren't for #778. Since at some point we want to toss out that `password_salt` column (or more likely, just stop using it for new passwords going forward) I'm not sure it's worth the effort to add the constraints.

Now that `password_salt` is nullable it might actually be pretty easy to implement #778